src/main/java/menu/TaskManagerCommands.java

```diff
@@ -16,17 +17,23 @@ public void showTask() {
         System.out.println("show");
     }
 
+    private Pattern onlyNumbers = Pattern.compile("[0-9]");
     public void removeTask() {
-//need validate input data
-        System.out.println("Which task's number do you wont to remove");
+        System.out.println("Which task's number do you want to remove");
         String input = scanConsoleInput();
-        int indexRemoveTask = Integer.parseInt(input) - 1;
-        tasks.remove(indexRemoveTask);
-        WriterTasks.serializeTasks(tasks);
-
-        System.out.println("remove");
+        if (input.matches(String.valueOf(onlyNumbers))){
+            int indexRemoveTask = Integer.parseInt(input) - 1;
+            try {
```

Comment:
It is bad practice to use try/catch inside if/else block. Better to create method that will be called inside if block

Reply:
> It is bad practice to use try/catch inside if/else block. Better to create method that will be called inside if block

Why is it a bad practice? Do you mean move the try-catch into a new method, and then call that method?

I know that it is [a bad practice to use exceptions handling as control flow](https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why)